orchestrator/dags/general_data_import.py

```diff
@@ -148,22 +146,21 @@ def extract_calendar_lines(general_file_name: str) -> List[Calendar]:
         for line in file:
             if "*KA" in line:
                 line = next(file)
-                f = open(f'{out_dir}calendar{current_date}.txt', "w", encoding="utf8")
-                while True:
-                    try:
-                        line_data = line.split()
-                        new_calendar_day = Calendar(line_data[0], line_data[2:])
-                        calendar_days.append(new_calendar_day)
-                        f.write(line)
-                        line = next(file)
-                    except StopIteration:
-                        # there is no lines left
-                        break
-                    if '#KA' in line:
-                        # we've reached the end of the data for given tram line
-                        f.close()
-                        break
-                continue
+                with open(f'{out_dir}calendar{current_date}.txt', "w", encoding="utf8") as f:
```

Comment:
Outfile is opened multiple times for each line in infile - the opening statement could be moved up, and placed just below opening statement of infile.